ui.apps/src/main/content/jcr_root/apps/venia/clientlibs/common/js/CommerceGraphqlApi.js

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ *
+ *     Copyright 2019 Adobe. All rights reserved.
+ *     This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License. You may obtain a copy
+ *     of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software distributed under
+ *     the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *     OF ANY KIND, either express or implied. See the License for the specific language
+ *     governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+
+(function () {
+
+    'use strict';
+    //todo update the proxy rules so that this gets router through the dispatcher
+    const imageUrlPrefix = "/magento/img";
+
+    class CommerceGraphqlApi {
+
+        constructor(props) {
+            if (!props.endpoint) {
+                throw new Error('The commerce API is not properly initialized. The "endpoint" property is missing from the initialization object');
+            }
+
+            this.endpoint = props.endpoint;
+        }
+
+        async _fetch(url, params) {
+            let response = await fetch(url, params);
+            if (!response.ok || response.errors) {
+                let message = await response.text();
+                throw new Error(message);
+            }
+            return response.json();
+        }
+
+        /**
+         * Retrieves the URL of the images for an array of product data
+         * @param productData a dictionary object with the following structure {productName:productSku}.
+         * The productName is used for filtering the query and the product SKU is used to identify the variant for which to retrieve the image
+         * @returns {Promise<any[]>}
+         */
+        async getProductImageUrls(productData) {
```

Comment:
Do you really need all the images just for the cart? I think it would be simpler to simply take the main thumbnail image, so you also don't need the image url prefix.

Reply:
WDYM by "main thumbnail image"?

Reply:
`productData.thumbnail.url`

Reply:
But this is the thumbnail of the product, not the variant, isn't it?